### DISCUSSION TITLE
Bitperfect with http output

### DISCUSSION THREAD

**Discussion by timsofteng** [MusicPlayerDaemon/MPD#1742] [General]
Hello. I would like to stream music from mpd to my phone without resempling. Is it possible? How to properly do it?

**Comment by MaxKellermann** [comment]
Don't. But if you insist, use a lossless encoder.

**Reply by timsofteng** [reply]
> Don't. But if you insist, use a lossless encoder.

What do you mean?

**Comment by skidoo23** [comment]
Define in mpd.conf a httpd audio output with lossless codec.

```
audio_output {
        type            "httpd"
        tags            "yes"
        name            "httpd on port 8001"
        always_on       "yes"
        encoder         "flac"
        port            "8001"
        max_clients     "0"                     # optional 0=no limit
}
```

I prefer runnig MPD in satellite mode on my Android smartphone. Just look here https://github.com/MusicPlayerDaemon/MPD/discussions/1656

**Reply by timsofteng** [reply]
> Define in mpd.conf a httpd audio output with lossless codec.
> 
> ```
> audio_output {
>         type            "httpd"
>         tags            "yes"
>         name            "httpd on port 8001"
>         always_on       "yes"
>         encoder         "flac"
>         port            "8001"
>         max_clients     "0"                     # optional 0=no limit
> }
> ```

What if music is in non flac fomat? Like .ape or whatever.

**Reply by timsofteng** [reply]
> I prefer runnig MPD in satellite mode on my Android smartphone. Just look here #1656

Do you mean this method?
https://github.com/MusicPlayerDaemon/MPD/discussions/1656#discussioncomment-4043186

**Comment by skidoo23** [comment]
If samplerate, number of bits per sample and number of channels are supported then decoding and re-encoding from and to a lossless format is transparent.

But Max comment is correct. You should avoid streaming and use satellite setup.

**Reply by timsofteng** [reply]
> If samplerate, number of bits per sample and number of channels are supported then decoding and re-encoding from and to a lossless format is transparent.
> 
> But Max comment is correct. You should avoid streaming and use satellite setup.

Will satellite setup work for iOS as well?
